framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1434,10 +1440,9 @@ private Object getValueOfExpressionInAnnotation(
         } else if (expr instanceof NullLiteralExpr) {
             stubWarn("Null found as value for %s. Null isn't allowed as an annotation value", name);
             return null;
-        } else {
-            stubWarn("Unexpected annotation expression: " + expr);
-            return null;
         }
+        stubWarn("Unexpected annotation expression: " + expr);
```

Comment:
Please don't make unrelated changes such as this; keep the diffs small.
If you want to make this change, please propose it in a different pull request.

Reply:
It's related

Reply:
Can you please explain how?  It would be helpful to also give a test case that fails without this change but succeeds with this change.

Reply:
Oh, I see now that you introduced fallthroughs above.  I think it's less confusing to handle them at the point.  Though in that particular case, I think you can replace parts of the code with a recursive call.  Thanks.

Reply:
addressed